advocacy_docs/supported-open-source/barman/pg-backup-api/03-tasks.mdx

```diff
@@ -20,68 +20,224 @@ tags:
+    {
+      "scope": "server",
+      "server_name": "my-server-a",
+      "archiver": "on",
+      "streaming_archiver": "off"
+    },
+    {
+      "scope": "server",
+      "server_name": "my-other-server-b",
+      "conninfo": "some_conn_info",
+      "streaming_conninfo": "some_streaming_conninfo",
+      "streaming_archiver": "on"
+    },
+    {
+      "scope": "model",
+      "model_name": "some-model",
+      "streaming_conninfo": "some_streaming_conn_info"
+    }
+  ]
+}
 ```
-{"operation_type": "recover",
- "backup_id": "20230221T155931",
- "remote_ssh_command": "ssh postgres@db_server_two.my.org",
- "destination_directory": "/var/lib/pgdata"}
+
+Assume the above JSON content is stored in a file named `payload-pg-backup-api.json`, you could issue a request to the API using `curl` in the following way:
+
+```bash
+curl -X POST http://barman-server.my.org/operations -H "content-type: application/json" -d @payload-pg-backup-api.json
+{
+  "operation_id": "20240118T104231"
+}
 ```
 
+The `operation_id` in the response is the ID of the operation that has been created in by the API.
+
+#### GET: /operations/OPERATION_ID
+
+This endpoint allows you to check the status of an instance level operation with ID `OPERATION_ID`, which can be either of these:
+
+ * `DONE`: if the operation finished successfully
+ * `FAILED`: if the operation finished with errors
+ * `IN_PROGRESS`: if the operation is still ongoing
+
+ Please find an example below:
+
+ ```bash
+ curl http://barman-server.my.org/operations/20240118T104231
+ {
+  "operation_id": "20240118T104231",
+  "status": "DONE"
+}
+ ```
+
+#### GET: /servers/NAME/operations
+
+Returns all server level tasks created by the API for Barman server `NAME`, if any.
+
+The following example shows the two tasks created on the `my-barman-server` Barman server.
+
+```bash
+curl http://barman-server.my.org/servers/my-barman-server/operations
+{
+  "operations": [
+    {
+      "id": "20240118T140951",
+      "type": "recovery"
+    },
+    {
+      "id": "20240118T141135",
+      "type": "config_switch"
+    }
+  ]
+}
+```
+
+#### POST: /servers/NAME/operations
+
+Request pg-backup-api to create server level operations.
+
+Currently pg-backup-api supports `recovery` and `config_switch` operations.
+
 !!! Note
-    You need to set the "content-type" header as "application/json" like the example below, otherwise you will receive a 400 Bad Request error.
+    You need to set the `content-type` header as `application/json` when executing `POST` requests to this endpoint, otherwise you will receive a `400 Bad Request error`.
+
+##### recovery
+
+This operation is used to request a remote execution of `barman recover` command.
+
+The JSON payload of the request must contain the following keys:
+
+ * `type`: set as `recovery` to request that operation;
+ * `backup_id`: ID of the backup to be restored from Barman server `NAME`. Supports whatever `barman recover` supports, including `latest` for example;
```

Comment:
That line "Supports whatever `barman recover` supports, including `latest` for example;"

I might re-phrase it into
You could use Barman shortcuts or anything that `barman recover` support, including `latest` for example;